spec/factories/project_steps.rb

```diff
@@ -0,0 +1,49 @@
+FactoryGirl.define do
+  factory :project_step do
+    # project_table 'Loan'
+    # project_id { create(:loan).id }
+    project { create(:loan) }
+    person
+    # need to make sure parent saved before assigning these
+    # summary { Faker::Lorem.sentence }
+    # details { Faker::Lorem.paragraph }
+    scheduled_date { Faker::Date.between(Date.civil(2004, 01, 01), Date.today) }
+    completed_date { Faker::Date.between(Date.civil(2004, 01, 01), Date.today) }
+    is_finalized [true,false].sample  ##JE todo: is ther a better pattern to assign a boolean?
+    type_option_id { ProjectStep::TYPE_OPTIONS.values.sample }
+
+    trait :completed do
+      completed_date { Faker::Date.between(date, Date.today) }
+    end
+
+    trait :past do
+      scheduled_date { Faker::Date.backward }
+    end
+
+    trait :future do
+      scheduled_date { Faker::Date.forward }
+    end
+
+    # is this even valid?
+    # trait :for_loan do
+    #   transient { loan_id 0 }
+    #   after(:build) do |event, evaluator|
```

Comment:
It doesn't actually set the loan's id to 0 there, a transient attribute like that is sort of a placeholder, you need to provide a default value, I chose zero, but what it does is allow you to pass an argument for a virtual attribute in that will be evaluated and processed, in this case a loan id. 

ProjectSteps were inherently for loans then as well, it was just more difficult to construct the association in the factories because it didn't follow the rails conventions, which required manually setting the tables and ids rather than leveraging a polymorphic association setter. Like I said, we don't need this anymore but it'd be good to take note of how the transient attributes are used because it's an incredibly useful pattern for creating temporary virtual attributes used in assigning other values.